src/main/java/com/airbnb/metrics/ParserForTagInMBeanName.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2015.  Airbnb.com
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.airbnb.metrics;
+
+import com.yammer.metrics.core.MetricName;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Pattern;
+
+/**
+ * Parser for kafka 0.8.2 or later version
+ * where the MBeanName contains tags and
+ * Scope will store tags as well.
+ */
+public class ParserForTagInMBeanName extends Parser {
+
+    public static final String SUFFIX_FOR_ALL = "_all";
+
+    @Override
+    public void parse(MetricName metricName) {
+        Pattern p = tagRegexMap.get(metricName.getType());
+        if (p != null && !p.matcher(metricName.getMBeanName()).matches()) {
+            name = sanitizeName(metricName, SUFFIX_FOR_ALL);
+        } else {
+            name = sanitizeName(metricName);
+        }
+        tags = getTags(metricName);
+    }
+
+    String[] getTags(MetricName metricName) {
```

Comment:
this method has the name has the `Parser#getTags` (in the super class). It's confusing.
Also let's make it `private` or `protected`